numpyro/distributions/copula.py

```diff
@@ -0,0 +1,74 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+from jax import lax, numpy as jnp
+
+import numpyro.distributions.constraints as constraints
+from numpyro.distributions.continuous import MultivariateNormal, Normal
+from numpyro.distributions.distribution import Distribution
+from numpyro.distributions.util import is_prng_key, validate_sample
+
+
+class GaussianCopula(Distribution):
+    arg_constraints = {
+        "correlation_matrix": constraints.corr_matrix,
+        "correlation_cholesky": constraints.lower_cholesky,
+    }
+    reparametrized_params = [
+        "correlation_matrix",
+        "correlation_cholesky",
+    ]
+
+    def __init__(
+        self,
+        marginal_dist,
+        correlation_matrix=None,
+        correlation_cholesky=None,
+        validate_args=None,
+    ):
+        self.marginal_dist = marginal_dist
+        self.base_dist = MultivariateNormal(
+            covariance_matrix=correlation_matrix,
+            scale_tril=correlation_cholesky,
+        )
+        self.normal = Normal()
+
+        event_shape = self.base_dist.event_shape
+        batch_shape = lax.broadcast_shapes(
+            self.marginal_dist.batch_shape[:-1],
+            self.base_dist.batch_shape,
+        )
```

Comment:
I think we can have
```
    if len(self.marginal_dist.batch_shape) > 0:
        batch_shape = lax.broadcast_shapes(
            self.marginal_dist.batch_shape[:-1],
            self.base_dist.batch_shape,
        )
    else:
        batch_shape = self.base_dist.batch_shape
```

Reply:
I'm not sure this check is necessary, in the case that `self.marginal_dist.batch_shape == ()` `self.marginal_dist.batch_shape[:-1]` still gives the correct value of `()`. My tests with `Beta(1.0, 2.0)` seem to behave correctly regardless of the event/batch shape of the base distribution.

Happy to make this change regardless if you feel that it should be included.

Reply:
Oh nice! I thought the slice op will give error - thanks